### PR TITLE
Catch 404 completed

### PR DESCRIPTION
**Pull request by coolalexzb**
Follow the commit in #1446.
The `urlretrieve` check solved.

### CODE REVIEW

**Comment by henrykironde** [comment]
```
{
    "citation": "R. A. Fisher. 1936. The Use of Multiple Measurements in Taxonomic Problems. and Asuncion, A. & Newman, D.J. (2007). UCI Machine Learning Repository [http://www.ics.uci.edu/~mlearn/MLRepository.html]. Irvine, CA: University of California, School of Information and Computer Science.",
    "description": "Famous dataset from R. A. Fisher. This dataset has been corrected. Information Source: Asuncion, A. & Newman, D.J. (2007). UCI Machine Learning Repository [http://www.ics.uci.edu/~mlearn/MLRepository.html]. Irvine, CA: University of California, School of Information and Computer Science.",
    "homepage": "http://mlr.cs.umass.edu/ml/datasets/Iris",
    "keywords": [
        "plants",
        "literature-compilation",
        "categorical"
    ],
    "name": "iris",
    "resources": [
        {
            "dialect": {
                "delimiter": ",",
                "header_rows": 0
            },
            "name": "Iris",
            "schema": {
                "fields": [
                    {
                        "name": "sepal_length",
                        "type": "double"
                    },
                    {
                        "name": "sepal_width",
                        "type": "double"
                    },
                    {
                        "name": "petal_length",
                        "type": "double"
                    },
                    {
                        "name": "petal_width",
                        "type": "double"
                    },
                    {
                        "name": "class",
                        "size": "20",
                        "type": "char"
                    }
                ]
            },
            "url": "https://swcarpentry.github.io/reference.html#database-manager"
        }
    ],
    "retriever": "True",
    "retriever_minimum_version": "2.0.dev",
    "title": "Iris Plants Database",
    "version": "1.2.1"
}

```

Use this script and try to see if it works. Paste the output if it works for you .

**Comment by henrykironde** [comment]
This is a changed version of iris and the url currently points to a 404 page

**Comment by henrykironde** [comment]
Also checkout the Pull request https://github.com/weecology/retriever/pull/1446
And from that, checkout a branch so that you add your changes on top of the old PR.
IF you need help with that ping me

**Comment by coolalexzb** [comment]
Hi, I merged branch in #1446 into my brach locally and repalced iris.json file with the content you mentioned above. I used `rt.download('iris')` to test and the result was following. It looks all good when catch 404 code.

屏幕快照 2020-03-22 下午9 54 48

**Comment by henrykironde** [comment]
Looks good to me. Thanks @coolalexzb I will wait for @apoorvaeternity to merge this.

File: retriever/lib/engine.py [hunk 1]
```diff
@@ -488,11 +490,21 @@ def download_file(self, url, filename):
                     hooks={'response': reporthook(progbar, path)},
                 )
 
+                if response.status_code == 404:
+                    print ("404 error: The data source not found, or server not found ")
```
Comment (apoorvaeternity) [inline retriever/lib/engine.py]: Remove the space after `print`.
This message seems more appropriate.
```
Error 404: The data source or server not found
```

Comment (coolalexzb) [reply retriever/lib/engine.py]: Removed! I will notice this small mistake next time.
Thx!

Comment (coolalexzb) [reply retriever/lib/engine.py]: Travis courrpted, but the code should be correct.

Comment (apoorvaeternity) [reply retriever/lib/engine.py]: @coolalexzb Can you also update the print message to `Error 404: The data source or server not found`

Comment (coolalexzb) [reply retriever/lib/engine.py]: Megered the master branch and updated the msg. Thx! @apoorvaeternity

**Pull request merged**